src/main/ruby/truffleruby/core/truffle/feature_loader.rb

```diff
@@ -198,11 +198,18 @@ def self.feature_provided?(feature, expanded)
     # MRI: loaded_feature_path
     # Search if $LOAD_PATH[i]/feature corresponds to loaded_feature.
     # Returns the $LOAD_PATH entry containing feature.
-    def self.loaded_feature_path(loaded_feature, feature, load_path)
+    def self.feature_path_loaded?(loaded_feature, feature, load_path)
       name_ext = extension(loaded_feature)
-      load_path.find do |p|
-        loaded_feature == "#{p}/#{feature}#{name_ext}" || loaded_feature == "#{p}/#{feature}"
-      end
+
+      path = if loaded_feature.end_with?("/#{feature}#{name_ext}")
+               loaded_feature.chomp("/#{feature}#{name_ext}")
+             elsif loaded_feature.end_with?("/#{feature}")
+               loaded_feature.chomp("/#{feature}")
+             else
+               return false
+             end
+
+      load_path.include?(path)
```

Comment:
I wonder if we could store the $LOAD_PATH index in the FeatureEntry, and use that as a faster check to avoid the loop (i.e. `load_path[index] == path or load_path.include?(path)`). OTOH we would need to care about $LOAD_PATH changing (tracked via version), in which case we should recompute the index for the FeatureEntry.

Actually, I think something simpler and easier would be to keep a `Hash` of the `$LOAD_PATH` (recomputed by version). And then we can do `load_path_hash.include?(path)` which seems really nice and simple.

Reply:
I believe something along that line can be accomplished. I'd rather ship this separately as that cached result change would still need some thinking how to position it properly. Would that work?

Reply:
Yeah, that's definitely something that can be done separately.